typings/index.d.ts

```diff
@@ -1418,21 +1418,15 @@ export class Message<Cached extends boolean = boolean> extends Base {
   public inGuild(): this is Message<true> & this;
 }
 
-export class MessageActionRow extends BaseMessageComponent {
-  public constructor(data?: MessageActionRow | MessageActionRowOptions | APIActionRowComponent);
+export type MessageActionRowComponentType<T> = T | MessageActionRowComponentResolvable[];
+
+export class MessageActionRow<T = MessageActionRowComponent> extends BaseMessageComponent {
```

Comment:
Sorry I didn't catch this earlier, but shouldn't this T be of type MessageComponentTypeResolvable just like in collectors? For consistency purposes that is

Reply:
No, mainly because the collectors accept the number or string representation of a component in its `options.componentType`. This only needs the `MessageActionRowComponent`, since it's the only type used in the class.

Reply:
Alright then this works just fine! Thanks